File: python/raindrops/raindrops.py
from collections import OrderedDict


def convert(num):
    # snd_dict = {3: "Pling", 5: "Plang", 7: "Plong"}
    snd_dict = OrderedDict([(3, "Pling"), (5, "Plang"), (7, "Plong")])
    snd = [snd_dict[k] for k in snd_dict if num % k == 0]
    if len(snd) == 0:
        return str(num)
    return "".join(snd)


# def convert(num):
#     snd_list = [(3, "Pling"), (5, "Plang"), (7, "Plong")]
#     snd = [note for factor, note in snd_list if num % factor == 0]
#     if len(snd) == 0:
#         return str(num)
#     return "".join(snd)

File: go/raindrops/raindrops.go
package raindrops

import "strconv"

var raindrops = [...]struct {
	factor int
	snd    string
}{
	{3, "Pling"},
	{5, "Plang"},
	{7, "Plong"},
}

// Convert turns a number into a string that contains raindrop sounds corresponding to certain potential factors
func Convert(num int) string {
	var numSnd string
	for _, raindrop := range raindrops {
		if (num % raindrop.factor) == 0 {
			numSnd += raindrop.snd
		}
	}
	if numSnd == "" {
		return strconv.Itoa(num)
	}
	return numSnd
}
